docs/sources/developers/plugins/plugin.schema.json

```diff
@@ -562,26 +569,103 @@
       }
     },
     "extensions": {
-      "type": "array",
-      "description": "List of link and component extensions which the plugin registers to given extension points.",
-      "items": {
-        "type": "object",
-        "properties": {
-          "extensionPointId": {
-            "type": "string"
-          },
-          "title": {
-            "type": "string"
-          },
-          "description": {
-            "type": "string"
-          },
-          "type": {
-            "type": "string",
-            "enum": ["link", "component"]
+      "type": "object",
+      "description": "...",
+      "properties": {
+        "addedComponents": {
+          "type": "array",
+          "description": "...",
+          "items": {
+            "type": "object",
+            "properties": {
+              "targets": {
+                "type": "array",
+                "description": "...",
+                "items": {
+                  "type": "string"
+                }
+              },
+              "title": {
+                "type": "string",
+                "description": "..."
+              },
+              "description": {
+                "type": "string",
+                "description": "..."
+              }
+            },
+            "required": ["targets", "title"]
```

Comment:
Even though it probably shouldn't be, `description` is currently required in the types for registering extensions. I'm happy to not enforce it on the meta info though.

Reply:
Good point! I was wondering if we could make it optional in the types for registration, too? (I don't think it should be required, right?)

Reply:
No I agree! (added a task in the [epic](https://github.com/grafana/grafana-community-team/issues/131) to change this)